temporal/api/cloud/cloudservice/v1/service.proto

```diff
@@ -125,4 +125,41 @@ service CloudService {
             get: "/api/v1/cloud/regions/{region}",
         };
     }
+
+       // Gets nexus endpoints
+    rpc GetNexusEndpoints(GetNexusEndpointsRequest) returns (GetNexusEndpointsResponse) {
+	option (google.api.http) = {
+            get: "/api/v1/nexus/endpoints",
+        };
+    }
+    
+    // Get a nexus endpoint
+    rpc GetNexusEndpoint(GetNexusEndpointRequest) returns (GetNexusEndpointResponse) {
+	option (google.api.http) = {
+            get: "/api/v1/nexus/endpoints/{endpoint_id}",
+        };
+    }
+
+    // Create a nexus endpoint
+    rpc CreateNexusEndpoint(CreateNexusEndpointRequest) returns (CreateNexusEndpointResponse) {
+        option (google.api.http) = {
+            post: "/api/v1/nexus/endpoints",
+	    body: "*"
+        };
+    }
+
+    // Update a nexus endpoint
+    rpc UpdateNexusEndpoint(UpdateNexusEndpointRequest) returns (UpdateNexusEndpointResponse) {
+	option (google.api.http) = {
+            post: "/api/v1/nexus/endpoints/{endpoint_id}",
```

Comment:
I almost missed this because it wasn't marked "ready for review" yet, heh.

We should combine these into one create-or-update, not sure the benefit of separating them. If you combine them you can use this URL, people expect POST to a resource like this to be able to create. If you must keep them separated, since we can't really support `PUT` accurately, I do think it deserves its own operation name.

Is there are good reason here or in operator service why we can't just have one mutation call?

Reply:
So few things here:
1) User and Namespace have separate create and update endpoints. We can definitely combine it for Nexus, but then that is inconsistent with the rest of our current Cloud APIs

2) Having distinction between Create/Update has been generally useful for us given that it gives flexibility for the operations to have different auth permissions (and is actually the case for namespaces for instance). Note that from the Cloud Side, create generates an ID that must be passed for Update (same for namespace / user - we take the IDs not the name/email address). We can scrap the ID of the "id" and only rely on the user provided field, but having this decoupled ID seems better in the long-term. 

3) Namespace and User have `POST` as well for update on the resource path itself. If we were to keep these separate, should we update the existing paths on those two types to have an `/update` suffix hanging off of the end as well? I would think so, but just double-checking.

cc @anekkanti to go over this as well

Reply:
Synced with @anekkanti offline - we will add "update" suffix to the other update operation paths here and keep CREATE/UPDATE as separate operations for the reasons stated above.

Reply:
Now that users can't provide their own ID, I don't mind if you leave it as is now FWIW. My primary concern was when they could provide their own ID, then the ID-based URL should be create-or-update.

Reply:
reverted